### PR TITLE
Refactor gle

### PR DESCRIPTION
**Pull request by R00ney**
refactored  opts.rs and windows.rs  to  create command line flag -G  [GL or ES2]    or  --graphics  [GL or ES2],   with error checking on  option. These flags then pass to rust-layers  to  select  GL or ES2 rendering (see servo/rust-layers#219)

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8790)

### CODE REVIEW

File: components/util/opts.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@ use url::{self, Url};
 pub struct Opts {
     pub is_running_problem_test: bool,
 
+
```
Comment (glennw) [inline components/util/opts.rs]: Not needed

**Comment by glennw** [comment]
Review status: 0 of 2 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[components/util/opts.rs, line 192 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aKTNmZN1Mn8B2icF6:-K4aKTNnd_Mp00PU2Yu7:-555043805) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/components/util/opts.rs#L192)):*
Let's add a comment here describing what this is for, and that it's only applicable on Linux.

---

*[components/util/opts.rs, line 406 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aKfppZa01ZOJdURjU:-K4aKfppZa01ZOJdURjV:1273876432) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/components/util/opts.rs#L406)):*
Now that we have switched the string to be stored as the enum type, this function does nothing and can be removed.

---

*[components/util/opts.rs, line 513 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aKrDUTiY0acpIkrBw:-K4aKrDUTiY0acpIkrBx:-1724225266) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/components/util/opts.rs#L513)):*
Indentation seems wrong here - tabs vs. spaces?

---

*[components/util/opts.rs, line 652 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aL1xixRr3zNwx0B4Y:-K4aL1xixRr3zNwx0B4Z:-1935950274) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/components/util/opts.rs#L652)):*
These strings should be compared ignoring case.

---

*[components/util/opts.rs, line 736 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aL8flybsI3plJtpfL:-K4aL8flybsI3plJtpfM:36652031) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/components/util/opts.rs#L736)):*
Indentation wrong here (tabs vs spaces?)

---

*[ports/glutin/window.rs, line 144 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aLK3CYI0ADWewMtfp:-K4aLK3CYI0ADWewMtfq:2048204766) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/ports/glutin/window.rs#L144)):*
There should be a separate function here for android (like there was previously) that always selects OpenGlEs 2.0 for android. The option selection should only apply to Linux.

---

*[ports/glutin/window.rs, line 643 \[r4\]](https://reviewable.io:443/reviews/servo/servo/8790#-K4aLVgJgHnfNkicdIwi:-K4aLVgK2YQGdnkKBLCa:-1677730488) ([raw file](https://github.com/servo/servo/blob/ab861ff60381bd0e6db5a84a02da9c416cb0bba1/ports/glutin/window.rs#L643)):*
Indentation seems wrong here again.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8790)*

**Comment by jdm** [comment]
Note: be sure to run `./mach test-tidy` and fix any errors it reports!

**Comment by R00ney** [comment]
Closing for new PR

**Pull request closed without merging**